katcp/test/test_resource.py

```diff
@@ -94,8 +94,10 @@ def test_wait(self):
         # wait for value and status
         waiting_value = 3
         waiting_status = 'warn'
+        waiting_condition = lambda reading: reading.value == waiting_value and \
+                                            reading.status == waiting_status
         self.io_loop.add_callback(DUT.set_value, 3, Sensor.WARN)
-        yield DUT.wait(waiting_value, waiting_status)
+        yield DUT.wait(waiting_condition, timeout=timeout)
```

Comment:
@theunsa, see also this test to set your mind at ease that status checking still works :-)